manager/burrow-mint/evm/native.go

```diff
@@ -24,8 +24,12 @@ import (
 
 var registeredNativeContracts = make(map[Word256]NativeContract)
 
-func RegisteredNativeContract(addr Word256) bool {
-	_, ok := registeredNativeContracts[addr]
+func RegisteredNativeContractAddress(address []byte) bool {
```

Comment:
We should work towards stronger typing, not looser typing;  `LeftPadWord256` should be called by caller of `RegisteredNativeContract(word256) bool`; so please revert and adjust two places where you call.